dipy/reconst/fwdti.py

```diff
@@ -121,16 +123,24 @@ def __init__(self, gtab, fit_method="NLS", *args, **kwargs):
 
         if not callable(fit_method):
             try:
-                fit_method = common_fit_methods[fit_method]
+                self.fit_method = common_fit_methods[fit_method]
             except KeyError as e:
                 e_s = '"' + str(fit_method) + '" is not a known fit '
                 e_s += "method, the fit method should either be a "
                 e_s += "function or one of the common fit methods"
                 raise ValueError(e_s) from e
-        self.fit_method = fit_method
+        else:
+            self.fit_method = fit_method
```

Comment:
this was not necessary

Reply:
After the application of decorator I received an error saying
```python

    def test_fwdti_singlevoxel():
        # Simulation when water contamination is added
        gtf = 0.44444  # ground truth volume fraction
        mevals = np.array([[0.0017, 0.0003, 0.0003], [0.003, 0.003, 0.003]])
        S_conta, peaks = multi_tensor(
            gtab_2s,
            mevals,
            S0=100,
            angles=[(90, 0), (90, 0)],
            fractions=[(1 - gtf) * 100, gtf * 100],
            snr=None,
        )
        fwdm = fwdti.FreeWaterTensorModel(gtab_2s, "WLS")
>       fwefit = fwdm.fit(S_conta)

dipy/reconst/tests/test_fwdti.py:100: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
dipy/reconst/multi_voxel.py:40: in new_fit
    return single_voxel_fit(self, data)
dipy/testing/decorators.py:210: in wrapper
    return convert_positional_to_keyword(func, args, kwargs)
dipy/testing/decorators.py:191: in convert_positional_to_keyword
    return func(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <dipy.reconst.fwdti.FreeWaterTensorModel object at 0x13e1ce110>
data = array([ 100.        ,   43.50442139,   13.18018944,   33.49286891,
         18.75861633,   22.62819702,   37.1749802 ,...   5.72778443,
         29.20758655,    5.17472072,   35.21098605,   27.02812095,
         20.99614616,    5.7284468 ]), mask = None, kwargs = {}, S0 = 100.0

    @multi_voxel_fit
    @warning_for_keywords()
    def fit(self, data, *, mask=None, **kwargs):
        """Fit method of the free water elimination DTI model class
    
        Parameters
        ----------
        data : array
            The measured signal from one voxel.
        mask : array
            A boolean array used to mark the coordinates in the data that
            should be analyzed that has the shape data.shape[:-1]
        """
        S0 = np.mean(data[self.gtab.b0s_mask])
>       fwdti_params = self.fit_method(
            self.design_matrix, data, S0, *self.args, **self.kwargs
        )
E       TypeError: 'str' object is not callable

dipy/reconst/fwdti.py:161: TypeError
```
in my opinion, the `self.fit_method` was being set as a string either 'NLS' or 'WLS' and it was not able to set a corresponding function from `common_fit_methods`. The `else` clause ensures that `self.fit_method` is always set to a callable object, and it's a function from `common_fit_methods`.

I might be wrong tho, I am not 100% sure.